### PR TITLE
Update our package's dependencies for v4

### LINKED ISSUE #164: Update to new cucumber package location for v7?
Will involve a change in cucumber package name among other things. See message at https://www.npmjs.com/package/cucumber.

### LINKED ISSUE #394: Update to latest stable npm version of puppeteer
We're way behind.

### PR DESCRIPTION
**Pull request by plocket**
[This is fairly complex to match up with the changes in our dependencies.]

I'm thinking this is just going to be for v4. Not bothering with this for v3 unless we absolutely have to since none of the vulnerabilities are severe. My current rationale is that the more work we do to maintain 3, the less work we can do getting v4 ready for release. Ready to hear opinions.

- Close #164, update cucumber to v7
- Prepare for v8 of cucumber because I won't remember it later
- Close #394, update puppeteer
- Update our version of node (and that of our action that we'll run for other people's libs). [Addresses #393 so we can use the suffolk npm org package.]
- Use `npm audit` to fix the remaining vulnerabilities (now 0)
- [Remove package.json as discussed in #489 to align our tests' behaviors with those of our users.]

### CODE REVIEW

File: lib/steps.js [hunk 1]
```diff
@@ -752,7 +769,7 @@ When(/I download "([^"]+)"$/, async (filename) => {
 //#####################################
 //#####################################
 
-When(/I set the name of ?(?:the var(?:iable)?)? "([^"]+)" to "([^"]+)"/, async (var_base, name_str) => {
+When(/I set the name of ?(?:the var(?:iable)?)? "([^"]+)" to "([^"]+)"/, { timeout: scope.timeout }, async (var_base, name_str) => {
```
Comment (plocket) [inline lib/steps.js:772]:
```suggestion
When(/I set the name of ?(?:the var(?:iable)?)? "([^"]+)" to "([^"]+)"/, { timeout: -1 }, async (var_base, name_str) => {
```

File: action.yml [hunk 2]
```diff
@@ -41,8 +41,9 @@ runs:
             "takedown": "node_modules/.bin/da-takedown"
           },
           "dependencies": {
-            "cucumber": "^6.0.5",
-            "docassemble-cucumber": "^3.0.0"
+            "colors": "1.4.0",
+            "@cucumber/cucumber": "^7.3.2",
```
Comment (plocket) [inline action.yml:45]: Freeze this for security

File: package.json [hunk 3]
```diff
@@ -34,13 +34,13 @@
...
     "cheerio": "^1.0.0-rc.5",
-    "cucumber": "^6.0.5",
     "dotenv": "^8.2.0",
-    "mocha": "^7.2.0",
-    "puppeteer": "^3.1.0",
+    "mocha": "^9.2.0",
+    "puppeteer": "^13.1.3",
     "qs": "^6.10.2",
     "sanitize-filename": "^1.6.3",
     "uuid": "^8.3.2"
```
Comment (plocket) [inline package.json:46]: Freeze these. Maybe new issue?

File: lib/steps.js [hunk 4]
```diff
@@ -623,8 +654,10 @@ Then(/I get the(?: page's)?(?: JSON)? var(?:iable)?s and val(?:ue)?s/i, async ()
...
   */
-
-  let data = await scope.getPageJSON( scope );
+  let data = await wrapPromiseWithTimeout(
+    scope.steps.json_vars( scope ),
+    scope.timeout
+  );
 
   // In future, may be saved to artifact file
   await scope.addToReport( scope, {
```
Comment (plocket) [inline lib/steps.js:663]: Remove this report

**Review by BryceStevenWilley — COMMENTED** [review]
LGTM with the already commented changes

**Review by BryceStevenWilley — APPROVED** [review]

**Pull request merged**